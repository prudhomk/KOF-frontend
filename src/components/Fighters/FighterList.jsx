import React from 'react';
import { Link } from 'react-router-dom';
import { useFighters } from '../../state/customHooks';
import Fighter from './Fighter';

const FighterList = () => {
  const { fighters, loading } = useFighters();
  if(loading) return <h1>Loading...</h1>;

  const fighterElements = fighters.map((fighter) => (
    <li key={fighter.id}>
      <Link to={`/${fighter.id}`}>
        <Fighter {...fighter} />
      </Link>
    </li>
  ));

  return (
  
    <ul>{fighterElements}</ul>
  );

};

export default FighterList;
