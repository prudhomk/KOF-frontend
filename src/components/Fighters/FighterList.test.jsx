/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FighterList from './FighterList';
import { MemoryRouter } from 'react-router-dom';

describe('Villager List component', () => {
  const list = [
    {
      name: 'hello',
      birthplace: 'USA',
      style: 'karate',
      image: 'whatever',
      _id: 'string'
    }
  ];
  afterEach(() => cleanup());
  it('renders Fighter List', () => {
    const { asFragment } = render(<MemoryRouter><FighterList fighters={list}/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
