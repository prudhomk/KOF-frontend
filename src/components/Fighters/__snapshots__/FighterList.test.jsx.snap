// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager List component renders Fighter List 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;
